pkg/trait/dependencies.go

```diff
@@ -50,7 +50,7 @@ func (t *dependenciesTrait) Apply(e *Environment) error {
 			util.StringSliceUniqueAdd(&dependencies, dep)
 		}
 	}
-	for _, s := range e.Integration.Spec.Sources {
+	for _, s := range append(e.Integration.Spec.Sources, e.Integration.Status.GeneratedSources...) {
```

Comment:
Not sure about this as it could eventually append GeneratedSources to Sources if the underlying slice has some capacity.

There is an utility method `Integration.Sources()` that should properly combine the two sources

Reply:
Thanks @lburgazzoli , fixed.

Reply:
there is a lint issue

    pkg/trait/dependencies_test.go:171: line is 178 characters